### PR TITLE
OPC UA terminate functionality

### PR DESCRIPTION
**Pull request by rahulp13**
### Purpose

Graceful termination of simulation through OPC UA

### Approach

- Add `terminate` variable to the OPC UA server's address space
- Do not print terminfo on termination
- Fixed triggering event on each step once after getting new input
- Wait for step after updating variables with the latest values

### CODE REVIEW

File: OMCompiler/SimulationRuntime/opc/ua/omc_opc_ua.c [hunk 1]
```diff
@@ -648,8 +689,6 @@ int omc_embedded_server_update(void *state_vp, double t)
   MODEL_DATA *modelData = data->modelData;
   int latestValues;
 
-  waitForStep(state);
```
Comment (sjoelund) [inline OMCompiler/SimulationRuntime/opc/ua/omc_opc_ua.c:651]: Any reason why waitForStep was moved? I think the original intent might have been to set variables before initialization (when paused).

**Comment by sjoelund** [comment]
It looks like some tests are affected by this: https://test.openmodelica.org/jenkins/blue/organizations/jenkins/OpenModelica/detail/PR-8310/1/tests
Maybe the indentation of some messages was changed by accident.

File: OMCompiler/SimulationRuntime/opc/ua/omc_opc_ua.c (hunk 1 above)
Comment (rahulp13) [reply OMCompiler/SimulationRuntime/opc/ua/omc_opc_ua.c:651]: The variables are updated after taking the current step and the next simulation loop is also done before waiting for the next step. So, it gives a false notion that the updated values are from the current simulation loop, but actually are from the previous one.

I'm not sure about `before initialization`. Does it mean before the start of the simulation main loop?

Comment (sjoelund) [reply OMCompiler/SimulationRuntime/opc/ua/omc_opc_ua.c:651]: Or no. Maybe it had to do with dassl in real-time mode. You get better real-time performance if you wait after the integrator runs. It's so long ago I forgot the reasons why :)

Comment (rahulp13) [reply OMCompiler/SimulationRuntime/opc/ua/omc_opc_ua.c:651]: Oh! ... Here, the wait is still after the integrator runs. It's that when to update the variables.

**Comment by adrpo** [comment]
@sjoelund can you have a quick look and merge this in if is fine?

**Comment by adrpo** [comment]
It seems this broke the Windows builds with:

`C:\OMDev\tools\msys\mingw64\bin\ld: omc_opc_ua.o:omc_opc_ua.c:(.rdata$.refptr.omc_terminate[.refptr.omc_terminate]+0x0): undefined reference to `omc_terminate'`

I guess one would need to dllimport this symbol for MinGW.

**Comment by adrpo** [comment]
Reverted (#8358) until we find a fix for Windows.

**Pull request merged**